### ISSUE TITLE
High loss in train_vae.py after replacing tg with predicted refractive index

### ISSUE THREAD

**Issue by fly-man-fly** [vashisth-lab/VitrimerVAE#3]
Hello! We are very interested in your project and have successfully reproduced your code, but we have a question. We are exploring additional physical properties of Vitrimer (e.g., refractive index) using your database. We replaced all the tg values in your labeled data with refractive index values predicted by a machine learning model and attempted to run the code. We executed the commands sequentially: python preprocess.py --ncpu 32 followed by python train_vae.py --savedir results.
We noticed that the loss value in the training log is much higher compared to when using the original tg values—specifically, the loss_epo hovers around 11. However, our understanding is that train_vae.py performs unsupervised training and should not rely on labeled data. Why then has the model performance degraded so significantly after replacing the data with refractive index values?
Attached below are the log files from the original code, our log files, and our raw dataset for your reference.

[8.log](https://github.com/user-attachments/files/24121311/8.log)
[mytest.csv](https://github.com/user-attachments/files/24121312/mytest.csv)
[myprop_train.csv](https://github.com/user-attachments/files/24121310/myprop_train.csv)
[my8.log](https://github.com/user-attachments/files/24121313/my8.log)

**Comment by yiwenzheng98** [comment]
Thanks for reporting the issue.

You are correct that unsupervised learning should not be affected by the label. My guess is that the loss was stuck in local mininum and could not be further optimized. If you didn't see a significant drop in loss_epo (down to ~1) in the first epoch, I would suggest changing the random seed and training the model again:

```bash
python train_vae.py --savedir results --seed 42
```

**Comment by fly-man-fly** [comment]
[0.log](https://github.com/user-attachments/files/24146891/0.log)I trained the model using your code with the command: python train_vae.py --savedir results --seed 42, but this time the loss_aci has been consistently high. What surprises me is that in train_vae.py, the line parser.add_argument('--seed', type=int, default=5) sets the default random seed to 5. I trained the model with both tg data and refractive index data using the random seed 5—so why are the results different?

**Comment by yiwenzheng98** [comment]
Could you share more details about how you obtained 8.log and my8.log? Did you run `python preprocess.py` twice, one for Tg and one for refractive index?

**Comment by fly-man-fly** [comment]
Yes, first I ran python preprocess.py and python train_vae.py using the original code. These scripts process the original tg data, resulting in the file 8.log. Then I copied the VAE folder for backup. After replacing the values in tg, I deleted the files generated by python preprocess.py and python train_vae.py, and then ran these two scripts in sequence again, which produced my8.log.

**Comment by yiwenzheng98** [comment]
Since the training data for `train_vae.py` is the same, I think you can proceed with `train_vae_prop.py` for refractive index data based on the model that produced 8.log. In the meantime, I will look into what caused this discrepancy and provide updates here.
